File: src/main/java/com/swp391/koibe/repositories/OtpRepository.java
package com.swp391.koibe.repositories;

import com.swp391.koibe.models.Otp;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface OtpRepository extends JpaRepository<Otp, Long> {

    Optional<Otp> findByEmail(String email);
    //findByEmailAndOtp(String email, int otp);
    Optional<Otp> findByEmailAndOtp(String email, String otp);

}

File: src/main/java/com/swp391/koibe/repositories/OtpRepository.kt
package com.swp391.koibe.repositories

import com.swp391.koibe.models.Otp
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import java.time.LocalDateTime
import java.util.*

interface OtpRepository : JpaRepository<Otp, Long> {
    fun findByEmailAndOtp(email: String, otp: String): Optional<Otp>
    @Modifying
    @Query("UPDATE Otp o SET o.isExpired = true WHERE o.expiredAt < :now AND o.isExpired = false")
    fun updateExpiredOtps(now: LocalDateTime?)
}
